pkg/pillar/hypervisor/kubevirt.go

```diff
@@ -486,46 +514,48 @@ func (ctx kubevirtContext) Start(domainName string) error {
 	if !ok {
 		return logError("start domain %s failed to get vmlist", domainName)
 	}
-	if vmis.mtype == IsMetaReplicaSet {
-		err := StartReplicaSetContiner(ctx, ctx.vmiList[domainName].replica)
+
+	// Start the Pod ReplicaSet
+	if vmis.mtype == IsMetaReplicaPod {
+		err := StartReplicaPodContiner(ctx, ctx.vmiList[domainName].repPod)
 		return err
+	} else if vmis.mtype != IsMetaReplicaVMI {
+		return logError("Start domain %s wrong type", domainName)
 	}
 
-	vmi := vmis.vmi
+	// Start the VMI ReplicaSet
+	repvmi := vmis.repVMI
 	virtClient, err := kubecli.GetKubevirtClientFromRESTConfig(kubeconfig)
-
 	if err != nil {
 		logrus.Errorf("couldn't get the kubernetes client API config: %v", err)
 		return err
 	}
 
-	// Create the VM
+	// Create the VMI ReplicaSet
 	i := 5
 	for {
-		_, err = virtClient.VirtualMachineInstance(kubeapi.EVEKubeNameSpace).Create(context.Background(), vmi)
+		_, err = virtClient.ReplicaSet(kubeapi.EVEKubeNameSpace).Create(repvmi)
```

Comment:
@naiming-zededa does this move to some default context?

Reply:
@andrewd-zededa, No, it's still in the 'eve-kube-app', is this you meant by context?

Reply:
no not namespace, the new Create() call removes context.Background() from the first arg

Reply:
I see on that version there is a default context.Background().
https://github.com/kubevirt/client-go/blob/09d24236db4ccb7a1872b8611d4baddd26a04ab3/kubecli/replicaset.go#L103

Reply:
the version we're on currently 1.2.0 https://github.com/naiming-zededa/eve/blob/01b9114736ccebbc607c221447b3f62e2d22b442/pkg/pillar/go.mod#L65

Reply:
well, they don't provide a way to pass in, so just default behavior of the context